src/main/webapp/app/shared/modal/AddMutationModal.tsx

```diff
@@ -80,507 +82,747 @@ function AddMutationModalContent({
-      };
+  function handleNormalAlterationChange(newValue: string, tabIndex: number) {
+    setTabStates(states => {
+      const newStates = _.cloneDeep(states);
+      newStates[tabIndex].alterationFieldValueWhileFetching = newValue;
+      return newStates;
     });
+    fetchNormalAlterationDebounced(newValue, tabIndex, tabStates);
+  }
 
-    function replaceOldMutations(oldMutationStates: MutationState[], newMutations: MutationState[], uuidToReplace: string) {
-      return oldMutationStates.reduce((accumulator: MutationState[], currentMutation) => {
-        if (uuidToReplace === currentMutation.uuid) {
-          return [...accumulator, ...newMutations];
-        }
-        return [...accumulator, currentMutation];
-      }, []);
+  async function fetchExcludedAlteration(
+    newAlteration: string,
+    tabIndex: number,
+    excludingIndex: number,
+    alterationData: AlterationData[]
+  ) {
+    const newParsedAlteration = parseAlterationName(newAlteration);
+
+    const currentState = alterationData[tabIndex];
+    const alteration = currentState.alteration.toLowerCase();
+    let excluding: string[] = [];
+    for (let i = 0; i < currentState.excluding.length; i++) {
+      if (i === excludingIndex) {
+        excluding.push(...newParsedAlteration.map(alt => alt.alteration.toLowerCase()));
+      } else {
+        excluding.push(currentState.excluding[excludingIndex].alteration.toLowerCase());
+      }
+    }
+    excluding = excluding.sort();
+    if (
+      alterationData.some(
+        state =>
+          state.alteration.toLowerCase() === alteration &&
+          _.isEqual(state.excluding.map(ex => ex.alteration.toLowerCase()).sort(), excluding)
+      )
+    ) {
+      notifyError(new Error('Duplicate alteration(s) removed'));
+      setTabStates(states => {
+        const newStates = _.cloneDeep(states);
+        newStates[tabIndex].excluding.splice(excludingIndex, 1);
+        return newStates;
+      });
+      return;
     }
 
-    setMutationStates(prev => {
-      if (excludingUuid) {
-        const newMutationStates = _.cloneDeep(prev);
-        const parentState = newMutationStates.find(state => state.uuid === uuid);
-        newAlterations.forEach(newAlteration => (newAlteration.Comment = null));
-        parentState.Excluding = replaceOldMutations(parentState.Excluding, newAlterations, excludingUuid);
-        return newMutationStates;
-      }
+    const newComment = newParsedAlteration[0].comment;
 
-      if (uuid) {
-        return replaceOldMutations(prev, newAlterations, uuid);
-      }
+    const alterationPromises: Promise<EntityStatusAlteration>[] = [];
+    let newAlterations: AlterationData[] = [];
+    if (newParsedAlteration[0].alteration !== alterationData[tabIndex]?.excluding[excludingIndex].alteration) {
+      alterationPromises.push(fetchAlteration(newParsedAlteration[0].alteration));
+    } else {
+      newAlterations.push(alterationData[tabIndex].excluding[excludingIndex]);
+    }
```

Comment:
I'm a bit lost here. Could you explain what's the logic here?

Reply:
I do the check to save a fetch for the following case. You start with the alteration `V600E`. You then change the alteration to `V600E/G`. Since the resulting alterations are `V600E` and `V600G` there is no need to refetch the first alteration since we already have it. In that case, I just added it to new alterations. 

If the alteration changed from `V600E/G` to `V600F/G` the logic in the if branch will execute since the first alteration would need to be refetched.